extras/tests/HelpTest/SimulatedStream.h

```diff
@@ -0,0 +1,90 @@
+#ifndef _TestStream_h
+#define _TestStream_h
+
+#include <Arduino.h>
+#include "../simpleSerialShellTest/simpleFIFO.h"
+
+/**
+ * @brief A simple implementation fo the Stream interface.  Useful for 
+ * creating mock Serial objects for unit testing.
+ * 
+ * A header-only class for ease of re-use across tests cases.
+ */
+template<int MAX_BUFSIZE> 
```

Comment:
First time I've had a template class in my own code base.  I guess I had to learn sometime....